solutions/src/test/java/ru/sberbank/school/task10/ConcurrentScalableThreadPoolTest.java

```diff
@@ -0,0 +1,169 @@
+package ru.sberbank.school.task10;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ConcurrentScalableThreadPoolTest {
+    private int max = 10;
+    private int core = 3;
+    private ThreadPool threadPool;
+
+    @Before
+    public void init() {
+        threadPool = new ConcurrentScalableThreadPool(core, max);
+    }
+
+    private Runnable newRunnable(int time, AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+            try {
+                TimeUnit.MILLISECONDS.sleep(time);
+            } catch (InterruptedException ignore) {
+//                        e.printStackTrace();
+            }
+            synchronized (counter) {
+                System.out.println("Task: " + counter.incrementAndGet()
+                        + " - Thread: " + Thread.currentThread().getName());
+            }
+            latch.countDown();
+        };
+    }
+
+    private Callable<String> newCallable(int time, AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+            TimeUnit.MILLISECONDS.sleep(time);
+            String res;
+            synchronized (counter) {
+                res = "Task: " + counter.incrementAndGet() + " - Thread: " + Thread.currentThread().getName();
+            }
+            latch.countDown();
+            return res + " " + LocalDateTime.now().toString();
+        };
+    }
+
+    @Test
+    public void normalWork() throws InterruptedException {
+        final AtomicInteger counter = new AtomicInteger(0);
+        Thread.sleep(10);
+        int initialThreadSize = Thread.getAllStackTraces().keySet().size();
+        CountDownLatch latch;
```

Comment:
Distance between variable &#39;latch&#39; declaration and its first usage is 5, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1815778488>*